### DISCUSSION TITLE
support automatic bucketed batching with dynamic batch sizes for multi-GPU

### DISCUSSION THREAD

**Discussion by wangleiofficial** [huggingface/datasets#5960] [Q&A]
I know the InfiniBatch should support this feature，can [datasets](https://github.com/huggingface/datasets) achieve?

**Comment by mariosasko** [comment]
Batching techniques such as InfiniBatch are out of the scope of this library.

**Reply by wangleiofficial** [reply]
Thanks. Can InfiniBatch be combined with datasets?
